src/renderer/src/routes/app/projects/$projectId_/settings/categories.tsx

```diff
@@ -0,0 +1,244 @@
+
+export const Route = createFileRoute(
+	'/app/projects/$projectId_/settings/categories',
+)({
+	loader: async ({ context, params }) => {
+		const { clientApi, queryClient } = context
+		const { projectId } = params
+
+		let projectApi
+		try {
+			// TODO: Not ideal but requires changes in @comapeo/core-react
+			projectApi = await queryClient.ensureQueryData({
+				queryKey: [COMAPEO_CORE_REACT_ROOT_QUERY_KEY, 'projects', projectId],
+				queryFn: async () => {
+					return clientApi.getProject(projectId)
+				},
+			})
+		} catch {
+			throw notFound()
+		}
+
+		// TODO: Not ideal but requires changes in @comapeo/core-react
+		queryClient.ensureQueryData({
+			queryKey: [
+				COMAPEO_CORE_REACT_ROOT_QUERY_KEY,
+				'projects',
+				projectId,
+				'project_settings',
+			],
+			queryFn: async () => {
+				return projectApi.$getProjectSettings()
+			},
+		})
+	},
+	component: RouteComponent,
+})
+
+const DEFAULT_CATEGORIES_NAME = `@mapeo/default-config`
+
+function RouteComponent() {
+	const { formatMessage: t, formatDate } = useIntl()
+	const router = useRouter()
+
+	const { projectId } = Route.useParams()
+
+	const { data: projectSettings } = useProjectSettings({ projectId })
+
+	const selectCategoriesFile = useMutation(
+		selectCategoriesFileMutationOptions(),
+	)
+
+	const importCategoriesFile = useImportProjectConfig({ projectId })
+
+	const someMutationPending =
+		selectCategoriesFile.status === 'pending' ||
+		importCategoriesFile.status === 'pending'
+
+	return (
+		<Stack direction="column" flex={1} overflow="auto">
+			<Stack
+				direction="row"
+				alignItems="center"
+				component="nav"
+				useFlexGap
+				gap={4}
+				padding={4}
+				borderBottom={`1px solid ${BLUE_GREY}`}
+			>
+				<IconButton
+					onClick={() => {
+						if (someMutationPending) {
+							return
+						}
+
+						if (router.history.canGoBack()) {
+							router.history.back()
+							return
+						}
+
+						router.navigate({
+							to: '/app/projects/$projectId/settings',
+							params: { projectId },
+							replace: true,
+						})
+					}}
+				>
+					<Icon name="material-arrow-back" size={30} />
+				</IconButton>
+
+				<Typography variant="h1" fontWeight={500}>
+					{t(m.navTitle)}
+				</Typography>
+			</Stack>
+
+			<Stack
+				direction="column"
+				flex={1}
+				justifyContent="space-between"
+				overflow="auto"
+				padding={6}
+				useFlexGap
+				gap={6}
+			>
+				<Stack
+					direction="column"
+					borderRadius={2}
+					border={`1px solid ${BLUE_GREY}`}
+					flex={1}
+					justifyContent="center"
+					useFlexGap
+					gap={5}
+					padding={6}
+				>
+					<Box alignSelf="center">
+						<Icon name="material-category" htmlColor={DARK_ORANGE} size={128} />
+					</Box>
+
+					<Typography variant="h1" fontWeight={500} textAlign="center">
+						{projectSettings.configMetadata?.name ||
+							// TODO: Get confirmation about what this should be
+							t(m.missingCategoriesName)}
```

Comment:
have some lingering doubts about what to do in this case. easy to address as a follow-up but also easy to completely forget about 😄